src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala

```diff
@@ -330,28 +327,44 @@ class GTGRunner(val masterURL: String,
       nodeMinArea)
 
 
+    edgeListRDD.cache()
+    edgeListRDD.localCheckpoint()
+
     /**
      * Collect the edgeList and construct NodeMap
      */
     val MCCEdgeList = edgeListRDD.collect()
     val MCCNodeMap = createNodeMapFromEdgeList(MCCEdgeList, lat, lon)
 
+    val broadcastedNodeMap = sc.sparkContext.broadcast(MCCNodeMap)
+
     /**
      * Process the edge list. Collect and output edges and vertices
      */
     processEdges(MCCEdgeList, MCCNodeMap)
 
-    val edgeListRDDIndexed = MCCOps.createPartitionIndex(edgeListRDD.collect())
-    val count = edgeListRDDIndexed.size
+    /**
+     * Generate the netcdfs
+     */
+    val edgeTupleRDD = edgeListRDD.foreach( x =>
```

Comment:
initializing an edgeTupleRDD variable was pointless here.
I should have removed this and just kept edgeListRDD.foreach(x => ....